zerocopy-derive/src/lib.rs

```diff
@@ -644,25 +717,23 @@ fn derive_from_zeros_enum(ast: &DeriveInput, enm: &DataEnum) -> proc_macro2::Tok
     // the allowed ones.
     try_or_print!(ENUM_FROM_ZEROS_INTO_BYTES_CFG.validate_reprs(ast));
 
-    let has_explicit_zero_discriminant =
-        enm.variants.iter().filter_map(|v| v.discriminant.as_ref()).any(|(_, e)| {
-            if let Expr::Lit(ExprLit { lit: Lit::Int(i), .. }) = e {
-                i.base10_parse::<usize>().ok() == Some(0)
-            } else {
-                false
-            }
-        });
-    // If the first variant of an enum does not specify its discriminant, it is set to zero:
-    // https://doc.rust-lang.org/reference/items/enumerations.html#custom-discriminant-values-for-fieldless-enumerations
-    let has_implicit_zero_discriminant =
-        enm.variants.iter().next().map(|v| v.discriminant.is_none()) == Some(true);
-
-    if !has_explicit_zero_discriminant && !has_implicit_zero_discriminant {
-        return Error::new_spanned(
-            ast,
-            "FromZeros only supported on enums with a variant that has a discriminant of `0`",
-        )
-        .to_compile_error();
+    if let Err(has_unknown_variants) = find_zero_variant(enm) {
+        if has_unknown_variants {
+            return Error::new_spanned(
+                ast,
+                "FromZeros only supported on enums with a variant that has a discriminant of `0`\n\
+                This enum has discriminants that the derive macro cannot evaluate (e.g. const \
```

Comment:
```suggestion
                This enum has discriminants that `#[derive(FromZeros)]` cannot evaluate (e.g. const \
```

Just to be friendly to people who aren't familiar with the terminology (ie, new Rustaceans who just saw that they needed to do this derive thing and don't know that they're technically called macros). I don't feel strongly about this suggestion - feel free to ignore.

Reply:
I like the wording change